File: string-handling/string-position/python/show_position.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0

def main():
    """Demonstrates finding a letter within a string"""
    subject = "Computer Science"
    letter_sought = "m"
    show_position(subject, letter_sought)
    position = subject.find(letter_sought)
    print(f"{letter_sought} found at: {position}")
    

# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: string-handling/string-position/visual-basic/show_position.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim subject As String = "Computer Science"
        Dim letterSought As Char = "m"
        Dim position As Integer = subject.IndexOf(letterSought)
        Console.WriteLine($"{letterSought} found at: {position}")
    End Sub


End Module
